packages/dd-trace/src/appsec/rule_manager.js

```diff
@@ -26,137 +22,66 @@ function loadRules (config) {
 }
 
 function updateWafFromRC ({ toUnapply, toApply, toModify }) {
-  const batch = new Set()
-
-  const newRulesData = new SpyMap(appliedRulesData)
-  let newRuleset
-  let newRulesetId
-  const newRulesOverride = new SpyMap(appliedRulesOverride)
-  const newExclusions = new SpyMap(appliedExclusions)
-  const newCustomRules = new SpyMap(appliedCustomRules)
   const newActions = new SpyMap(appliedActions)
 
+  let wafUpdated = false
+  let wafUpdatedSuccess = true
+
   for (const item of toUnapply) {
-    const { product, id } = item
+    if (!ASM_PRODUCTS.has(item.product)) continue
+
+    try {
+      waf.wafManager.remove(item.path)
+      item.apply_state = ACKNOWLEDGED
+      wafUpdated = true
 
-    if (product === 'ASM_DATA') {
-      newRulesData.delete(id)
-    } else if (product === 'ASM_DD') {
-      if (appliedRulesetId === id) {
-        newRuleset = defaultRules
+      if (item.product === 'ASM') {
+        newActions.delete(item.id)
       }
```

Comment:
isn't there a way to get the actions from the WAF instead of doing the state managemnt ourselves ?